### ISSUE TITLE
example_collector_udp_test.go fails with seg fault. 

### ISSUE THREAD

**Issue by AbhishekRamamurthy** [zoomoid/go-ipfix#1]
Stack:

require github.com/zoomoid/go-ipfix v0.4.0

```
on 0.0.0.0:2055
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x761c6a]

goroutine 36 [running]:
github.com/zoomoid/go-ipfix.(*Message).Decode(0x0, {0x8f61a0, 0xc0002184e0})
	/root/go/pkg/mod/github.com/zoomoid/go-ipfix@v0.4.0/message.go:86 +0x8a
github.com/zoomoid/go-ipfix.(*Decoder).Decode(0xc00021e000, {0x8f8360, 0xc000168a50}, 0xc0002184e0)
	/root/go/pkg/mod/github.com/zoomoid/go-ipfix@v0.4.0/decode.go:118 +0x14a
main.main.func3()
	/root/test/test-go-ipfix/main.go:83 +0x1da
created by main.main in goroutine 1
	/root/test/test-go-ipfix/main.go:79 +0x48c
```

**Comment by AbhishekRamamurthy** [comment]
(*Message).Decode() receiver is un-initialized after addressing this issue, field cache gets disconnected at field cache was initialized during NewDecoder init. 

```
func (d *TemplateSet) Decode(r io.Reader) (n int, err error) {
	d.Records = make([]TemplateRecord, 0)
	// "as long as there's set header data (Set ID, Length)"
	for {
		templateRecord := TemplateRecord{}

		m, err := templateRecord.Decode(r)
```


```
root@x:~/test# ./test
2024/02/29 20:20:00 Starting UDP listener for IPFIX messages on 0.0.0.0:2055
Abhishek: templateId: 256, fieldCount: 10
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x657938]

goroutine 22 [running]:
test/go-ipfix.(*TemplateRecord).decodeTemplateField(0xc000073b80, {0x792ac0, 0xc00009c150})
	/root/test/go-ipfix/template_record.go:173 +0x238
test/go-ipfix.(*TemplateRecord).Decode(0xc000073b80, {0x792ac0, 0xc00009c150})
	/root/test/go-ipfix/template_record.go:120 +0x227
test/go-ipfix.(*TemplateSet).Decode(0x6a5640?, {0x792ac0, 0xc00009c150})
	/root/test/go-ipfix/sets.go:261 +0x90
test/go-ipfix.(*Decoder).Decode(0xc000160d80, {0x7940f0, 0xc000322050}, 0xc00009c0c0)
	/root/test/go-ipfix/decode.go:155 +0x78c
main.main.func3()
	/root/test/udp_main.go:51 +0x1da
created by main.main in goroutine 1
	/root/test/udp_main.go:47 +0x48c
root@x:~/test# vi go-ipfix/template_record.go
```

**Comment by zoomoid** [comment]
Hey, thanks for reporting this, I had a look at your segfaults and both should be addressed by the v0.4.1 release, although I currently lack time and also data to test this out more thoroughly. Feel free to open new issues or re-open this one and I'll try my best to resolve any issues.

Note that I'm currently not actively working with IPFIX and this library, so my responses might be a bit delayed.
